pkg/registry/core/pod/strategy.go

```diff
@@ -133,6 +133,9 @@ func (podStrategy) CheckGracefulDelete(ctx context.Context, obj runtime.Object,
 		// use the default value if set, or deletes the pod immediately (0)
 		if pod.Spec.TerminationGracePeriodSeconds != nil {
 			period = *pod.Spec.TerminationGracePeriodSeconds
+			if period < 0 {
+				period = 0
+			}
```

Comment:
shouldn't this be moved to line 141 so it happens regardless of whether the value came from the request or the pod spec?